src/main/java/org/folio/service/job/JobExecutionServiceImpl.java

```diff
@@ -49,25 +53,31 @@
   }
 
   @Override
-  public Future<Optional<JobExecution>> getById(final String jobExecutionId, final String tenantId) {
-    return jobExecutionDao.getById(jobExecutionId, tenantId);
+  public Future<JobExecution> getById(final String jobExecutionId, final String tenantId) {
+    return jobExecutionDao.getById(jobExecutionId, tenantId)
+      .compose(optionalJobExecution -> {
+        if (optionalJobExecution.isPresent()) {
+          return succeededFuture(optionalJobExecution.get());
+        } else {
+          String errorMessage = String.format("Job execution not found with id %s", jobExecutionId);
+          LOGGER.error(errorMessage);
+          return failedFuture(new NotFoundException(errorMessage));
+        }
+      });
   }
 
   @Override
   public void updateJobStatusById(String id, JobExecution.Status status, String tenantId) {
-    getById(id, tenantId).compose(optionalJobExecution -> {
-      optionalJobExecution.ifPresent(jobExecution -> {
+    getById(id, tenantId).onSuccess(jobExecution -> {
```

Comment:
what happens if the update fails?

Reply:
@KVupp we just logging the error on the DAO level